Whiskey/Functions/Remove-WhiskeyFileSystemItem.ps1

```diff
@@ -31,49 +31,63 @@ function Remove-WhiskeyFileSystemItem
+        Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
     }
-    elseif( (Test-Path -Path $Path -PathType Container) )
+
+    Process
     {
-        if( $IsWindows )
+        if( (Test-Path -Path $Path -PathType Leaf ) )
+        {
+            Remove-Item -Path $Path -Force
+        }
+        elseif( (Test-Path -Path $Path -PathType Container) )
         {
-            $logPath = Join-Path -Path ([IO.Path]::GetTempPath()) -ChildPath ('whiskey.robocopy.{0}.log' -f ([IO.Path]::GetRandomFileName()))
-            $emptyDir = Join-Path -Path ([IO.Path]::GetTempPath()) -ChildPath ([IO.Path]::GetRandomFileName())
-            $deleteLog = $true
-            New-Item -Path $emptyDir -ItemType 'Directory' | Out-Null
-            try
+            if( $IsWindows )
             {
-                Invoke-WhiskeyRobocopy -Source $emptyDir -Destination $Path -LogPath $logPath | Out-Null
-                if( $LASTEXITCODE -ge 8 )
+                $logPath = Join-Path -Path ([IO.Path]::GetTempPath()) -ChildPath ('whiskey.robocopy.{0}.log' -f ([IO.Path]::GetRandomFileName()))
+                $emptyDir = Join-Path -Path ([IO.Path]::GetTempPath()) -ChildPath ([IO.Path]::GetRandomFileName())
+                $deleteLog = $true
+                New-Item -Path $emptyDir -ItemType 'Directory' | Out-Null
+                try
                 {
-                    $deleteLog = $false
-                    Write-Error -Message ('Failed to remove directory "{0}". See "{1}" for more information.' -f $Path,$logPath)
-                    return
+                    Invoke-WhiskeyRobocopy -Source $emptyDir -Destination $Path -LogPath $logPath | Out-Null
+                    if( $LASTEXITCODE -ge 8 )
+                    {
+                        $deleteLog = $false
+                        Write-Error -Message ('Failed to remove directory "{0}". See "{1}" for more information.' -f $Path,$logPath)
+                        return
+                    }
+                    Remove-Item -Path $Path -Recurse -Force
                 }
-                Remove-Item -Path $Path -Recurse -Force
-            }
-            finally
-            {
-                if( $deleteLog )
+                finally
                 {
-                    Remove-Item -Path $logPath -ErrorAction Ignore -Force
+                    if( $deleteLog )
+                    {
+                        Remove-Item -Path $logPath -ErrorAction Ignore -Force
+                    }
+                    Remove-Item -Path $emptyDir -Recurse -Force
                 }
-                Remove-Item -Path $emptyDir -Recurse -Force
+            }
+            else
+            {
+                Remove-Item -Path $Path -Recurse -Force
             }
         }
         else
         {
-            Remove-Item -Path $Path -Recurse -Force
+            Write-Error -Message ('Could not find "{0}" anywhere within parent directory' -f $Path) -ErrorAction Continue

```

Comment:
Remove this block. If a path doesn't exist, don't fail, assume it was already deleted, and happily return without doing any work.